src/libraries/System.Net.Http/src/System/Net/Http/HttpTelemetry.cs

```diff
@@ -0,0 +1,100 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics.Tracing;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace System.Net.Http
+{
+    [EventSource(Name = "System.Net.Http")]
+    internal sealed class HttpTelemetry : EventSource
+    {
+        public static readonly HttpTelemetry Log = new HttpTelemetry();
+
+        private IncrementingPollingCounter? _requestsPerSecondCounter;
+        private PollingCounter? _startedRequestsCounter;
+        private PollingCounter? _currentRequestsCounter;
+        private PollingCounter? _abortedRequestsCounter;
+
+        private long _startedRequests;
+        private long _currentRequests;
+        private long _abortedRequests;
+
+        public static new bool IsEnabled => Log.IsEnabled();
+
+        // NOTE
+        // - The 'Start' and 'Stop' suffixes on the following event names have special meaning in EventSource. They
+        //   enable creating 'activities'.
+        //   For more information, take a look at the following blog post:
+        //   https://blogs.msdn.microsoft.com/vancem/2015/09/14/exploring-eventsource-activity-correlation-and-causation-features/
+        // - A stop event's event id must be next one after its start event.
+
+        [Event(1, Level = EventLevel.Informational)]
+        public void RequestStart(string host, int port, string scheme, string path, string version)
+        {
+            Interlocked.Increment(ref _startedRequests);
+            Interlocked.Increment(ref _currentRequests);
+            WriteEvent(1, host, port, scheme, path, version);
```

Comment:
Does this bind to `WriteEvent(id, params object[])`? 

I assume we need a custom overload here to remove the allocations, as done in [NetEventSource](https://github.com/dotnet/runtime/blob/bab2bc4bfaaae52860e8f6f8ef1770c1030e70f1/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs#L505).

Reply:
Yes, I'll add a custom overload and add it to this PR.

In a future PR, we'll refactor all the custom overloads of `WriteEvent` from https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/Net/Logging/NetEventSource.Common.cs and place all of them in a single file.